### DISCUSSION TITLE
api routes 405 error.... please help me....

### DISCUSSION THREAD

**Discussion by kysclient** [vercel/next.js#52226] [App Router]
I'm using next.js 13.4.7 right now and the api route that works fine in my local environment is spitting 405 in vercel production mode. who can solve it?

**here is my client code** 

 const response = await fetch(endpoint, {
                method: 'POST',
                headers: {
                    'Content-Type': 'application/json',
                },
                signal: controller.signal,
                body,
            });
            console.log('response : ', response)

**in prod console :** 

body: ReadableStream
bodyUsed: false
headers: 
Headers {}
ok: false
redirected: false
status: 405
statusText: ""
type: "basic"

**and here is my api route code**

export const config = {
    runtime: 'edge',
};

const handler = async (req: Request): Promise<Response> => { } 

**and here is my vercel.json code**

{
  "routes": [
    {
      "src": "/api/aichat",
      "dest": "/api/aichat.ts"
    }
  ]
}

i also have env variables in my vercel dashboard..  

I tried everything I could. what's left?

**Comment by masterbater** [comment]
> I'm using next.js 13.4.7 right now and the api route that works fine in my local environment is spitting 405 in vercel production mode. who can solve it?
> 
> **here is my client code**
> 
> const response = await fetch(endpoint, { method: 'POST', headers: { 'Content-Type': 'application/json', }, signal: controller.signal, body, }); console.log('response : ', response)
> 
> **in prod console :**
> 
> body: ReadableStream bodyUsed: false headers: Headers {} ok: false redirected: false status: 405 statusText: "" type: "basic"
> 
> **and here is my api route code**
> 
> export const config = { runtime: 'edge', };
> 
> const handler = async (req: Request): Promise => { }
> 
> **and here is my vercel.json code**
> 
> { "routes": [ { "src": "/api/aichat", "dest": "/api/aichat.ts" } ] }
> 
> i also have env variables in my vercel dashboard..
> 
> I tried everything I could. what's left?

405 means method not allowed, your api might have been GET request than POST.

**Reply by kysclient** [reply]
no.............you can see my client fetch code. I send POST and also I can found method POST in my console... now i'm testing my app, I found my problem that can not access any api routes. it become always 405 error. what can I do ??

**Reply by icyJoseph** [reply]
You opened a new issue for the same problem?

You are using Pages Directory right? Or did you use app dir? I forgot to ask that in last question (and now we have two threads for the same error)

**Reply by masterbater** [reply]
https://nextjs.org/docs/app/building-your-application/routing/router-handlers#supported-http-methods

**Comment by Elindo586** [comment]
Did you fix this one?  What did you do?
